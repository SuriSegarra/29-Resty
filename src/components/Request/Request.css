.Request {

    input[type='text'] {
        position: relative;
        bottom: 695px;
        left: 28%;
        width: 50%;
        height:20px;
        /* margin: 1rem 0; */
    }

    input[type='radio'] {
        background-color: initial;
        -webkit-appearance: radio;
        box-sizing: border-box;
        margin: 3px 0.5ex;
        padding: initial;
        border: initial;
        position: relative;
        bottom: 650px;
        right: 20%;
        
    }
    input[type="radio"]:checked + label {
        background-color: #2d5986;
        color: white;
        
      }
label {
    padding: .25em 1em;
    margin: 0.5rem 0rem;
    cursor: pointer;
    bottom: 650px;
    position: relative;
    right: 20%;
    
    }
    textarea {
        position: relative;
        bottom: 580px;
        left: 29%;
        display: block;
        height: 10em;
        width: 40%;
    }
    
    button {
        position: relative;
        bottom: 835px;
        left: 80%;
        width: 30px;
        height: 20px;
    }
}