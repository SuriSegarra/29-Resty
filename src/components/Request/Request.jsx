import React from 'react';
import PropTypes from 'prop-types';

const Request  = ({ url, method, body, onChange, onSubmit }) => (
  <form onSubmit={onSubmit}>
    <input type='text' name='url' value={url} onChange={onChange} />
    
    <label> 
      <input type='radio' name='method' value='POST' checked={method === 'POST'} onChange={onChange}/>
      POST
    </label>

    <label>
      <input type='radio' name='method' value='GET' checked={method === 'GET'} onChange={onChange}/>
      GET
    </label>

    <label>
      <input type='radio' name='method' value='PUT' checked={method === 'PUT'} onChange={onChange}/>
      PUT
    </label>

    <label>
      <input type='radio' name='method' value='PATCH' checked={method === 'PATCH'} onChange={onChange}/>
      PATCH
    </label>

    <label>
      <input type='radio' name='method' value='DELETE' checked={method === 'DELETE'} onChange={onChange}/>
      DELETE
    </label>

    <textarea name='body' value={body} onChange={onChange}> </textarea>
    <button>Go!</button>
  </form>
);

Request.propTypes = {
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default Request;

