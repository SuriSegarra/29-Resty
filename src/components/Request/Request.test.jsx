  
import React from 'react';
import { shallow } from 'enzyme';
import Request from './Request';

describe('Request concomponent', () => {
  it('matches a snapshot of Request.jsx', () => {
    const wrapper = shallow(<Request
      url='test url'
      method='GET'
      body='JSON body'
      onChange={()=>{}}
      onSubmit={()=>{}}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
