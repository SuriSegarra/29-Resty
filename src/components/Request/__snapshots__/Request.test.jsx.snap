// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request concomponent matches a snapshot of Request.jsx 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="url"
    onChange={[Function]}
    type="text"
    value="test url"
  />
  <label>
    <input
      checked={false}
      name="method"
      onChange={[Function]}
      type="radio"
      value="POST"
    />
    POST
  </label>
  <label>
    <input
      checked={true}
      name="method"
      onChange={[Function]}
      type="radio"
      value="GET"
    />
    GET
  </label>
  <label>
    <input
      checked={false}
      name="method"
      onChange={[Function]}
      type="radio"
      value="PUT"
    />
    PUT
  </label>
  <label>
    <input
      checked={false}
      name="method"
      onChange={[Function]}
      type="radio"
      value="PATCH"
    />
    PATCH
  </label>
  <label>
    <input
      checked={false}
      name="method"
      onChange={[Function]}
      type="radio"
      value="DELETE"
    />
    DELETE
  </label>
  <textarea
    name="body"
    onChange={[Function]}
    value="JSON body"
  >
     
  </textarea>
  <button>
    Go!
  </button>
</form>
`;

exports[`Request matches Request jsx 1`] = `
<form
  className="Request"
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value="https://hey-arnold-api.herokuapp.com/api/v1/characters"
  />
  <input
    id="GET"
    name="method"
    onChange={[Function]}
    type="radio"
    value="GET"
  />
  <label
    htmlFor="GET"
  >
    GET
  </label>
  <input
    id="POST"
    name="method"
    onChange={[Function]}
    type="radio"
    value="POST"
  />
  <label
    htmlFor="POST"
  >
    POST
  </label>
  <input
    id="PUT"
    name="method"
    onChange={[Function]}
    type="radio"
    value="PUT"
  />
  <label
    htmlFor="PUT"
  >
    PUT
  </label>
  <input
    id="PATCH"
    name="method"
    onChange={[Function]}
    type="radio"
    value="PATCH"
  />
  <label
    htmlFor="PATCH"
  >
    PATCH
  </label>
  <input
    id="DELETE"
    name="method"
    onChange={[Function]}
    type="radio"
    value="DELETE"
  />
  <label
    htmlFor="DELETE"
  >
    DELETE
  </label>
  <textarea
    onChange={[Function]}
    value="json body"
  >
     
  </textarea>
  <button>
    Go!
  </button>
</form>
`;
